deepspeed/runtime/zero/partition_parameters.py

```diff
@@ -332,11 +466,79 @@ def _post_init_method(self, module):
 
     def _set_dtype(self, ds_config, dtype):
         if ds_config is not None and dtype is None:
-            self.dtype = torch.half if ds_config.fp16_enabled else torch.float
-        elif dtype is None:
-            self.dtype = torch.half
+            if ds_config.bfloat16_enabled and ds_config.fp16_enabled:
+                raise RuntimeError("bfloat16 and fp16 cannot be enabled at once")
+
+            if ds_config.bfloat16_enabled:
+                self.dtype = torch.bfloat16
+            elif ds_config.fp16_enabled:
+                self.dtype = torch.half
+            else:
+                self.dtype = torch.float
         else:
-            self.dtype = dtype
+            self.dtype = dtype or torch.half
+
+
+class AllGatherHandle:
+    def __init__(self, handle, param: Parameter) -> None:
+        self.__handle = handle
+        self.__param = param
```

Comment:
Is it useful to assert that ```param.ds_status == ZeRoParamStatus.INFLIGHT```, like is done in `AllGatherCoalescedHandle`?
```

Reply:
sure thats worth adding

Reply:
added